src/v/cloud_io/basic_cache_service_api.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "base/units.h"
+
+#include <seastar/core/file.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/io_priority_class.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/lowres_clock.hh>
+
+#include <filesystem>
+#include <iterator>
+#include <optional>
+#include <set>
+#include <string_view>
+
+namespace cloud_io {
+
+static constexpr size_t default_write_buffer_size = 128_KiB;
```

Comment:
you want `inline constexpr` if it is in header file scope; https://www.youtube.com/watch?v=QVHwOOrSh3w

Reply:
yeh. we are getting ready to enable clang tidy in bazel which is going to more or less enforce this in most cases

Reply:
https://github.com/redpanda-data/redpanda/pull/23810